src/EditorFeatures/Core/Interactive/InteractiveSession.cs

```diff
@@ -194,6 +199,18 @@ private void AddSubmissionProjectNoLock(ITextBuffer submissionBuffer, string lan
             _currentSubmissionProjectId = ProjectId.CreateNewId(newSubmissionProjectName);
             var newSubmissionDocumentId = DocumentId.CreateNewId(_currentSubmissionProjectId, newSubmissionProjectName);
 
+            // Associate a name with both the editor document and our roslyn document so LSP can make requests on it.
+            var extension = languageName switch
+            {
+                LanguageNames.CSharp => "cs",
+                LanguageNames.VisualBasic => "vb",
+                _ => throw new ArgumentException($"Unexpected language name {languageName}")
+            };
+            var submissionFilePath = Path.Combine(Path.GetTempPath(), $"Submission{SubmissionCount}.{extension}");
+
+            _textDocumentFactoryService.TryGetTextDocument(submissionBuffer, out var textDocument);
+            textDocument.Rename(submissionFilePath);
```

Comment:
@tmat  It might be possible to move this change to the interactive window repo so that we don't have to rename the document here.  However from testing this appears to work (we now get text sync notifications with the appropriate name and contents).  Wanted to check with you if you'd prefer to see this code somewhere else.